commcare_connect/opportunity/tasks.py

```diff
@@ -63,7 +65,33 @@ def add_connect_users(user_list: list[str], opportunity_id: str):
             username=user.username, defaults={"phone_number": user.phone_number, "name": user.name}
         )
         opportunity_access, _ = OpportunityAccess.objects.get_or_create(user=u, opportunity_id=opportunity_id)
-        invite_user(u, opportunity_access)
+        invite_user.delay(u.id, opportunity_access.id)
+
+
+@celery_app.task()
+def invite_user(user_id, opportunity_access_id):
```

Comment:
nit: user_id is not required since the user is already a part of the opportunity_access and the data for the user can be resolved from the `opportunity_access.user` property.